db/00_init.sql

```diff
@@ -254,3 +254,76 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql
 IMMUTABLE LEAKPROOF ROWS 100;
+
+-- Statistics per project and administrative boundary
+CREATE TABLE pdm_feature_counts_per_boundary(
+	project VARCHAR NOT NULL,
+	boundary BIGINT NOT NULL,
+	day DATE NOT NULL,
+	amount INT NOT NULL,
+
+	CONSTRAINT pdm_feature_counts_per_boundary_pk PRIMARY KEY(project, boundary, day)
+);
+
+CREATE INDEX pdm_feature_counts_per_boundary_project_idx ON pdm_feature_counts_per_boundary(project);
+
+-- Count edits per administrative boundary
+CREATE OR REPLACE FUNCTION pdm_add_changes_per_boundary(project_id VARCHAR, start_ts TIMESTAMP, end_ts TIMESTAMP) RETURNS VOID AS $$
```

Comment:
Pourquoi déclarer une procédure au lieu de mettre les requetes dans 12_projects_contribs.sql ?

Reply:
Pour pouvoir boucler sur les dates, je ne crois pas que ce soit possible de faire un LOOP hors d'une fonction PLPGSQL

Reply:
Non en effet pas hors d'une fonction pgsql, mais il faut intégrer ca dans le script `10_project_update.js`

On y traite déjà les changement sans avoir besoin de faire la suppression et une boucle pour itérer sur les dates.

Si tu veux je peux le faire

Reply:
Ça me semblait plus clair de mettre ça dans une fonction, mais si tu penses que c'est aussi bien dans `10_project_update.js` tu peux l'y intégrer oui :wink: